api/src/modules/Session/resolvers.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  mutationField,
+  nonNull,
+  stringArg,
+  queryField,
+  list,
+} from 'nexus/dist';
+
+export const listSessions = queryField('sessions', {
+  type: list('Session'),
+  resolve: async (_, __, ctx) => {
+    const { id } = await ctx.getUser();
+    return ctx.prisma.session.listSessions(id);
+  },
+});
+
+export const revoke = mutationField('revokeSession', {
+  type: 'Session',
+  args: {
+    sessionId: nonNull(stringArg()),
+  },
+  resolve: async (_parent, { sessionId }, ctx) => {
+    const userId = (await ctx.getUser()).id;
+    return ctx.prisma.session.revoke(sessionId, userId);
+  },
+});
+
+export const create = mutationField('createSession', {
+  type: 'SessionCreateOutput',
+  args: {
+    session: nonNull('SessionCreate'),
+  },
+  resolve: async (
+    _parent,
+    { session }: { session: { referenceExpiryDate?: Date; name: string } },
```

Comment:
Why does the `{ session }` is not properly typed according to `SessionCreate`? Does it mean the input is not validated?

Reply:
[According to this comment](https://github.com/makerdao-ses/switchboard-boilerplate/pull/28#discussion_r1151487407), please:

- [ ] Don't call whatever you receive here `referenceExpiryDate`
- [ ] Call it `expiresInSeconds`
- [ ] Receive `number`

Reply:
`{ session: { referenceExpiryDate?: Date; name: string } }` for some reason did not produce a warning with the mismatch (i.e. did not suggest to do `unknown as ...`). Just removing this one already checks the boxes.

Reply:
Can you please answer to the two original questions, because they might help me understand what happened here?
- Why does the { session } is not properly typed according to SessionCreate?
- Does it mean the input is not validated?

Reply:
> Should i deep dive into nexus docs for specific reason / look for TS rule so that this does not happen?

My question was, to reiterate, why it was needed to retype those types in the first place? How did `{ session }: { session: { referenceExpiryDate?: Date; name: string } }` happened in the first place? What type `session` have inside the resolver? `any` or is it properly typed according to the nexus input?

Reply:
> My question was, to reiterate, why it was needed to retype those types in the first place? 

this was probably an automatic instinct to type the variable for explicitness. There was no strong reason behind it for sure

> What type session have inside the resolver? any or is it properly typed according to the nexus input?

see the image - seems properly typed according to the nexus.

Image might look confusing since the popup is not directly visible. will add code snippet with typing hint below the image

![image](https://user-images.githubusercontent.com/59374892/228823547-873d0e57-9dcc-44bf-800a-44a7ef091bf0.png)

```
(parameter) session: {
    expiryDurationSeconds?: number | null | undefined;
    name: string;
}
```

Reply:
Thanks!